File: backend/app/schemas/show.py
from datetime import datetime
from typing import List, Literal, Optional
from pydantic import BaseModel, ConfigDict, Field

SectionType = Literal["featured", "series", "minisodes", "songs"]
StatusType = Literal["draft", "published"]


class CategoryRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    slug: str


class ShowCreate(BaseModel):
    title: str = Field(..., min_length=1)
    slug: str = Field(..., min_length=1)
    section: Optional[SectionType] = None
    description: Optional[str] = None
    status: StatusType = "draft"


class ShowUpdate(BaseModel):
    title: Optional[str] = Field(None, min_length=1)
    slug: Optional[str] = Field(None, min_length=1)
    section: Optional[SectionType] = None
    description: Optional[str] = None
    status: Optional[StatusType] = None


class ShowResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    title: str
    slug: str
    section: Optional[str] = None
    description: Optional[str] = None
    status: str
    created_at: datetime
    updated_at: datetime
    categories: List[CategoryRead] = []


class ShowListResponse(BaseModel):
    total: int
    page: int
    page_size: int
    results: List[ShowResponse]

File: cms/src/types/show.ts
export type SectionType = 'featured' | 'series' | 'minisodes' | 'songs';
export type StatusType = 'draft' | 'published';

export interface CategoryRead {
  id: number;
  name: string;
  slug: string;
}

export interface Show {
  id: number;
  title: string;
  slug: string;
  section: SectionType | null;
  description: string | null;
  status: StatusType;
  created_at: string;
  updated_at: string;
  categories: CategoryRead[];
}

export interface ShowListResponse {
  total: number;
  page: number;
  page_size: number;
  results: Show[];
}

export interface ShowCreatePayload {
  title: string;
  slug: string;
  section?: SectionType | null;
  description?: string | null;
  status?: StatusType;
}

export interface ShowUpdatePayload {
  title?: string;
  slug?: string;
  section?: SectionType | null;
  description?: string | null;
  status?: StatusType;
}

export interface ShowQueryParams {
  q?: string;
  section?: string;
  status?: string;
  page?: number;
  page_size?: number;
}
